components/login-register/register/pages/profile-type-page/profile-type-page.tsx

```diff
@@ -10,35 +11,79 @@ import { InnerFormWrapper } from '@/components/login-register/common/FormWrapper
 import { FormWrapper } from '../../../common/FormWrapper/form-wrapper'
 import { CheckBox } from '../../../common/Input/checkbox'
 import { useRegisterCtx } from '../../register-context'
+import { IProfileTypeInput, profileTypeFormSchema } from '../../schema'
 import { getProfileTypeIcon } from './helpers'
 
 export const ProfileTypePage = () => {
   const {
-    state: { profileType },
+    state: { profileType: ctxProfileType, customProfileType },
     setProfileType,
     setNextPageIndex,
+    setCustomProfileType,
   } = useRegisterCtx()
-  const [selectedProfileType, setSelectedProfileType] = useState(profileType)
   const { data: profileTypes, isLoading } = useProfileTypesList()
 
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    setError,
+    formState: { errors },
+  } = useForm<IProfileTypeInput>({
+    resolver: yupResolver(profileTypeFormSchema),
+    mode: 'onSubmit',
+    reValidateMode: 'onChange',
+    defaultValues: {
+      profileType: ctxProfileType,
+      otherObligationsValue: customProfileType,
```

Comment:
maybe something like this ?
```suggestion
      customProfileTypeName: customProfileType,
```

Reply:
XDD